crates/net/network/src/network.rs

```diff
@@ -227,14 +228,17 @@ impl PeersInfo for NetworkHandle {
 
 #[async_trait]
 impl NetworkInfo for NetworkHandle {
-    type Error = oneshot::error::RecvError;
+    type Error = Error;
```

Comment:
we should even get rid of this type and instead expect all functions to return `Resutl<T, NetworkError>` instead

Reply:
Removing type Error here also means removing it from ```NetworkInfo```.
https://github.com/paradigmxyz/reth/blob/009d2056f95c6541b71eff386ee1477de4ebbf3b/crates/net/network-api/src/lib.rs#L17-L28

Trying to use the ```NetworkError``` enum but I can't get it to be accessible in ```network-api```, adding ```reth-interfaces = { path = "../../interfaces" }``` to its cargo.toml, throws a cyclic dependency error.

Reply:
I feel like NetworkError should be declared in network-api instead

and `interfaces` should consume it